dotnet/targets/Xamarin.Shared.Sdk.targets

```
@@ -1022,6 +1048,39 @@
 		</ItemGroup>
 	</Target>
 
+	<Target Name="_CreateAOTDedupAssembly"
+			Condition="'$(_RunAotCompiler)' == 'true'">
```

```suggestion
	<Target Name="_CreateAOTDedupAssembly"
			Condition="'$(_RunAotCompiler)' == 'true'"
			DependsOnTargets="_ComputeManagedAssemblyToLink"
            BeforeTargets="PrepareForILLink"
    >
```
As you rightfully suggested, using the linker custom step `ComputeAOTArguments` would probably be the way to go. In order to do so, `_DedupAssembly` would have to be passed to the ILLinker, which can be achieved by injecting this target before executing the linker and adjusting the relevant ItemGroups